libpldm/pdr.c

```diff
@@ -64,12 +73,10 @@ static void add_hotplug_record(pldm_pdr *repo, pldm_pdr_record *record,
 		// printf("repo-last=%x, curr=%x, curr-next=%x",(unsigned
 		// int)repo->last, (unsigned int)curr, (unsigned
 		// int)curr->next);
-		record->next = curr->next;
-		curr->next = record;
-		if (record->next == NULL) {
-			repo->last->next = record;
-			repo->last = record;
-		}
+	}
+	if (recordAdded == false) {
+		free(record);
+		assert(recordAdded != false);
```

Comment:
We are going to force assert ie PLDM core here?

Reply:
Yes, we are asserting in the case where we fail to add a record